File: app/src/main/java/info/nightscout/androidaps/utils/XdripCalibrations.java
package info.nightscout.androidaps.utils;

import android.content.Context;
import android.content.Intent;
import android.content.pm.ResolveInfo;
import android.os.Bundle;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;

import info.nightscout.androidaps.Constants;
import info.nightscout.androidaps.MainApp;
import info.nightscout.androidaps.R;
import info.nightscout.androidaps.plugins.configBuilder.ProfileFunctions;
import info.nightscout.androidaps.services.Intents;

/**
 * Created by mike on 10.02.2017.
 */

public class XdripCalibrations {
    private static Logger log = LoggerFactory.getLogger(XdripCalibrations.class);

    public static void confirmAndSendCalibration(final Double bg, final Context parentContext) {
        if (parentContext != null) {
            String confirmMessage = String.format(MainApp.gs(R.string.send_calibration), bg);
            OKDialog.showConfirmation(parentContext, confirmMessage, () -> sendIntent(bg));
        }
    }

    public static boolean sendIntent(Double bg) {
        Context context = MainApp.instance().getApplicationContext();
        Bundle bundle = new Bundle();
        bundle.putDouble("glucose_number", bg);
        bundle.putString("units", ProfileFunctions.getSystemUnits().equals(Constants.MGDL) ? "mgdl" : "mmol");
        bundle.putLong("timestamp", System.currentTimeMillis());
        Intent intent = new Intent(Intents.ACTION_REMOTE_CALIBRATION);
        intent.putExtras(bundle);
        intent.addFlags(Intent.FLAG_INCLUDE_STOPPED_PACKAGES);
        context.sendBroadcast(intent);
        List<ResolveInfo> q = MainApp.instance().getApplicationContext().getPackageManager().queryBroadcastReceivers(intent, 0);
        if (q.size() < 1) {
            ToastUtils.showToastInUiThread(MainApp.instance().getApplicationContext(), MainApp.gs(R.string.xdripnotinstalled));
            log.debug(MainApp.gs(R.string.xdripnotinstalled));
            return false;
        } else {
            ToastUtils.showToastInUiThread(MainApp.instance().getApplicationContext(), MainApp.gs(R.string.calibrationsent));
            log.debug(MainApp.gs(R.string.calibrationsent));
            return true;
        }
    }
}

File: app/src/main/java/info/nightscout/androidaps/utils/XdripCalibrations.kt
package info.nightscout.androidaps.utils

import android.content.Context
import android.content.Intent
import android.os.Bundle
import info.nightscout.androidaps.Constants
import info.nightscout.androidaps.R
import info.nightscout.androidaps.logging.AAPSLogger
import info.nightscout.androidaps.interfaces.ProfileFunction
import info.nightscout.androidaps.services.Intents
import info.nightscout.androidaps.utils.resources.ResourceHelper
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class XdripCalibrations @Inject constructor(
    private val aapsLogger: AAPSLogger,
    private val resourceHelper: ResourceHelper,
    private val context: Context,
    private val profileFunction: ProfileFunction
) {

    fun sendIntent(bg: Double): Boolean {
        val bundle = Bundle()
        bundle.putDouble("glucose_number", bg)
        bundle.putString("units", if (profileFunction.getUnits() == Constants.MGDL) "mgdl" else "mmol")
        bundle.putLong("timestamp", System.currentTimeMillis())
        val intent = Intent(Intents.ACTION_REMOTE_CALIBRATION)
        intent.putExtras(bundle)
        intent.addFlags(Intent.FLAG_INCLUDE_STOPPED_PACKAGES)
        context.sendBroadcast(intent)
        val q = context.packageManager.queryBroadcastReceivers(intent, 0)
        return if (q.size < 1) {
            ToastUtils.showToastInUiThread(context, resourceHelper.gs(R.string.xdripnotinstalled))
            aapsLogger.debug(resourceHelper.gs(R.string.xdripnotinstalled))
            false
        } else {
            ToastUtils.showToastInUiThread(context, resourceHelper.gs(R.string.calibrationsent))
            aapsLogger.debug(resourceHelper.gs(R.string.calibrationsent))
            true
        }
    }
}